packages/clerk-js/src/ui/elements/FormControl.tsx

```diff
@@ -53,6 +64,57 @@ const getInputElementForType = (type: FormControlProps['type']) => {
   return CustomInputs[customInput] || Input;
 };
 
+function useFormTextAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
+    (enterAnimation: boolean): ThemableCssProp => {
+      if (prefersReducedMotion) {
+        return {
+          animation: 'none',
+        };
+      }
+      return t => ({
+        animation: `${enterAnimation ? animations.inAnimation : animations.outAnimation} ${
+          t.transitionDuration.$slower
+        } ${t.transitionTiming.$common}`,
+        transition: `height ${t.transitionDuration.$slow}  ease`, // This is expensive but required for a smooth layout shift
+      });
+    },
+    [prefersReducedMotion],
+  );
+
+  return {
+    getFormTextAnimation,
+  };
+}
+
+type Px = number;
+const useCalculateErrorTextHeight = () => {
+  const [height, setHeight] = useState<Px>(24);
+
+  const calculateHeight = useCallback((element: HTMLElement | null, messageToDisplay: string | undefined) => {
+    if (element) {
+      const fontSize = parseInt(getComputedStyle(element).fontSize.replace('px', ''));
+      const width = parseInt(getComputedStyle(element).width.replace('px', ''));
+      const lineHeight = parseInt(getComputedStyle(element).lineHeight.replace('px', '')) / 16;
+      const characters = messageToDisplay?.length || 0;
+
+      setHeight(prevHeight => {
+        const newHeight = 10 + fontSize * lineHeight * Math.ceil(characters / (width / (fontSize * 0.6))); //0.6 is an average character width
```

Comment:
❓ What does `10` represent?

Reply:
That refers to the 10px margin-top a form field text has by default